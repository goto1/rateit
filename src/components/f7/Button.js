import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const btnColorPicker = type => {
  switch (type) {
    case "primary":
      return {
        bgColor: "#057AFF",
        borderColor: "#057AFF",
        textColor: "#F1F1F5"
      };
    case "success": {
      return {
        bgColor: "#369947",
        borderColor: "#369947",
        textColor: "#F1F1F5"
      };
    }
    default: {
      return {
        bgColor: "inherit",
        borderColor: "#057AFF",
        textColor: "#057AFF"
      };
    }
  }
};

const Wrapper = styled.a`
  background-color: ${props => props.bgColor} !important;
  border-color: ${props => props.borderColor} !important;
  color: ${props => props.textColor} !important;
  text-transform: uppercase;
  letter-spacing: .5px;
`;

const Button = ({ type, onClick, children }) => {
  const { bgColor, borderColor, textColor } = btnColorPicker(type);
  return (
    <Wrapper
      href="#"
      onClick={onClick}
      className="button button-big"
      bgColor={bgColor}
      borderColor={borderColor}
      textColor={textColor}
    >
      {children}
    </Wrapper>
  );
};

Button.propTypes = {
  type: PropTypes.oneOf(["default", "primary", "success"]),
  onClick: PropTypes.func.isRequired,
  children: PropTypes.string.isRequired
};

export default Button;
