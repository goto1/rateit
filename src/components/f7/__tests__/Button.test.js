import React from "react";
import Button from "../Button";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";
import "jest-styled-components";

const props = {
  onClick: jest.fn()
};

describe("Button component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Button {...props}>Submit</Button>).toJSON();
    expect(tree).toMatchStyledComponentsSnapshot();
  });

  it("renders with the `default` color scheme", () => {
    const newProps = { ...props };
    const tree = renderer
      .create(<Button {...newProps}>Submit</Button>)
      .toJSON();
    expect(tree).toHaveStyleRule("background-color", "inherit !important");
    expect(tree).toHaveStyleRule("border-color", "#057AFF !important");
    expect(tree).toHaveStyleRule("color", "#057AFF !important");
  });

  it("renders with the `primary` color scheme", () => {
    const newProps = { ...props, type: "primary" };
    const tree = renderer
      .create(<Button {...newProps}>Submit</Button>)
      .toJSON();
    expect(tree).toHaveStyleRule("background-color", "#057AFF !important");
    expect(tree).toHaveStyleRule("border-color", "#057AFF !important");
    expect(tree).toHaveStyleRule("color", "#F1F1F5 !important");
  });

  it("renders with the `success` color scheme", () => {
    const newProps = { ...props, type: "success" };
    const tree = renderer
      .create(<Button {...newProps}>Submit</Button>)
      .toJSON();
    expect(tree).toHaveStyleRule("background-color", "#369947 !important");
    expect(tree).toHaveStyleRule("border-color", "#369947 !important");
    expect(tree).toHaveStyleRule("color", "#F1F1F5 !important");
  });

  it("calls a function when clicked", () => {
    const { onClick } = props;
    const wrapper = shallow(<Button {...props}>Submit</Button>);
    wrapper.simulate("click");
    expect(onClick.mock.calls.length).toBe(1);
  });
});
