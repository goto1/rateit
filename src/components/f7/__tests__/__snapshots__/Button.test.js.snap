// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
.ddvIbK {
  background-color: inherit !important;
  border-color: #057AFF !important;
  color: #057AFF !important;
  -webkit-text-transform: uppercase;
  text-transform: uppercase;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
}

<a
  className="button button-big sc-bdVaJa ddvIbK"
  href="#"
  onClick={[Function]}
>
  Submit
</a>
`;
